CorsixTH/Lua/dialogs/confirm_dialog.lua

```diff
@@ -37,26 +41,30 @@ function UIConfirmDialog:UIConfirmDialog(ui, text, callback_ok, callback_cancel)
   self:setDefaultPosition(0.5, 0.5)
   self.panel_sprites = app.gfx:loadSpriteTable("QData", "Req04V", true)
   self.white_font = app.gfx:loadFont("QData", "Font01V")
-  self.text = text
   self.callback_ok = callback_ok  -- Callback function to launch if user chooses ok
   self.callback_cancel = callback_cancel -- Callback function to launch if user chooses cancel
+  self.text = text
+
+  local maxLineLengthPixels = 152 -- this should come from somewhere else
+
+  -- Text is varable width, calculate it
+  local lineWidthPixels, textHeight = self.white_font:sizeOf(self.text)
+  local textRowHeight = textHeight + 1
 
-  -- Check how "high" the dialog must be
-  local w, h = self.white_font:sizeOf(text)
+  -- Calculate how many text rows we need
+  -- Retail has a fixed size box - but current behaviour is to calculate size of box
+  local totalLines = math.ceil(lineWidthPixels / maxLineLengthPixels)
```

Comment:
Old code did this but it's totally unnecessary and probably wrong. Just pass the max width as a second argument to font:sizeOf, and it will give you the total height of the resulting textbox.

Reply:
Which would probably fix the discrepancy between different dialogs, because actual line breaking would be taken into consideration...